### DISCUSSION TITLE
Manually uploaded Azure artifacts issue

### DISCUSSION THREAD

**Discussion by adjurdjevic** [rfennell/AzurePipelines#1960] [Q&A] [answered]
Hi
I have Classic Release pipeline which deploy couple of Universal Package from Azure Artifact together with a couple of artifacts which are associated with build pipelines. For this manually uploaded artifacts i receive a error in release generation task 

Looking at artifact [App]
Artifact type [Build]
Build Definition ID [undefined]
Build Number: [undefined]
Is Primary: [undefined]
Looking for the [App] in the most recent successful release [TEST 26.08.2024 - Rel. 3]
Found artifact [App] with build number [undefined] in release [TEST 26.08.2024 - Rel. 3]
Build for artifact [AppCosmoEdi] has not changed.  Nothing to do
Getting test associated with the latest build [undefined]
The most common runtime reason for the task to fail is due API ECONNRESET issues. To avoid this failing the pipeline these will be treated as warnings and an attempt to generate any release notes possible
Error: {"count":1,"value":{"Message":"The request is invalid."}}
    at RestClient.<anonymous> (/home/vsts/work/_tasks/XplatGenerateReleaseNotes_7b0a5887-75c4-4ade-a915-f92a93c934cb/4.17.12/node_modules/typed-rest-client/RestClient.js:202:31)
    at Generator.next (<anonymous>)
    at fulfilled (/home/vsts/work/_tasks/XplatGenerateReleaseNotes_7b0a5887-75c4-4ade-a915-f92a93c934cb/4.17.12/node_modules/typed-rest-client/RestClient.js:6:58)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  statusCode: 400,
  result: { count: 1, value: { Message: 'The request is invalid.' } },
Total Builds: [0]
  responseHeaders: {
Total Commits: [0]
    'cache-control': 'no-cache',
Total Workitems: [0]
    pragma: 'no-cache',
Total Related Workitems (Parent/Children): [0]
    'content-length': '57',
Total Release Tests: [0]
    'content-type': 'application/json; charset=utf-8; api-version=7.2-preview.1',
Total Tests: [0]
    expires: '-1',
Total Manual Test Runs: [0]
    p3p: 'CP="CAO DSP COR ADMa DEV CONo TELo CUR PSA PSD TAI IVDo OUR SAMi BUS DEM NAV STA UNI COM INT PHY ONL FIN PUR LOC CNT"',
Total Manual Test Configurations: [0]
    'x-tfs-processid': '32f6eb8f-3a50-44ea-bd9e-0504db94ce69',
Total Pull Requests: [0]
    'strict-transport-security': 'max-age=31536000; includeSubDomains',
Total Indirect Pull Requests: [0]
    activityid: 'e341d50c-27bd-472c-9c6a-7b0713b6da4e',
Total Associated Test WI: [0]
    'x-tfs-session': 'e341d50c-27bd-472c-9c6a-7b0713b6da4e',
Total Published Artifacts: [0]
    'x-vss-e2eid': 'e341d50c-27bd-472c-9c6a-7b0713b6da4e',
Total Consumed Artifacts: [0]
    'x-vss-senderdeploymentid': 'd057ee96-f15e-77ba-a4b5-e816729e2236',
Total WIQL Workitems: [0]
    'x-vss-userdata': '92b75838-cb07-4268-a6c6-bb9571de8496:3e3db4cc-de53-4570-8b24-93448280c3e1',
    'x-frame-options': 'SAMEORIGIN',
    'request-context': 'appId=cid-v1:75f7e064-d741-4620-89f0-b5624ba3fe6d',
    'access-control-expose-headers': 'Request-Context',
    'x-content-type-options': 'nosniff',
    'x-cache': 'CONFIG_NOCACHE',
    'x-msedge-ref': 'Ref A: 4683510C9D8C43EAAFBC1BF5FCC4C8C9 Ref B: DUS30EDGE0406 Ref C: 2024-08-26T14:54:35Z',
    date: 'Mon, 26 Aug 2024 14:54:34 GMT',
    connection: 'close'
  }
}

After this error further generation of release note stop and task produce file without any work items/commits in it. 
Can this problematic artifacts be somehow skipped or not included to avoid this error and to continue generation for the rest of artifacts.

Thank you

**Comment by adjurdjevic** [comment]
I found out that this issue is probably related to issue #937. When I remove this function and run testconsole without it, release notes get generated. Is there a way to create variable for this fix as you do for fix 349?

Thx

**Reply by rfennell** [reply]
I think the issue is that it is trying to get release notes for an artifact that is not a build (hence all the API calls fail). I suspect it only needs a guard clause to make sure it skips over this type of resource.

I have converted this thread to an issue to track getting the work done, not sure when it will be but it is on the backlog, but happy to look at PRs

**Comment by rfennell** [comment, marked as answer]
Fix released as 4.18.4
